source/model/backtraceMetrics.ts

```diff
@@ -0,0 +1,242 @@
+    this.sessionId = this.getSessionId();
+    this.lastActive = this.getLastActive();
+
+    this.persistSession(); // Create/persist session on construction.
+
+    // Get current session interval. If one is set or running, clear it.
+    const currentIntervalId = this.getActiveSessionIntervalId();
+    if (currentIntervalId) {
+      clearInterval(currentIntervalId);
+    }
+    // Start new interval.
+    // Persist session if page is focused on heartbeat interval
+    const intervalId = setInterval(
+      () => this.persistIfFocused(),
+      this.heartbeatInterval,
+    );
+    this.setActiveSessionIntervalId(intervalId);
+  }
+
+  /**
+   * Handle persisting of session. When called, will create or manage current session.
+   * when appropriate.
+   */
+  private persistSession(): void {
+    // If sessionId is not set, create new session. Send unique and app launch events.
+    if (!this.sessionId) {
+      this.createNewSession();
+      this.sendUniqueEvent();
+      // An "application launch" loosely / temporarily means first session creation.
+      this.sendSummedEvent('Application Launches');
+
+      // If sessionId is set and lastActive is over persistenceInterval, create new session and send unique event.
+      // if lastActive is not defined, the page was just launched; use current timestamp as lastActive.
+    } else if (
+      (this.lastActive || this.timestamp) <
+      this.timestamp * SEC_TO_MILLIS - this.persistenceInterval
+    ) {
+      this.createNewSession();
+      this.sendUniqueEvent();
+    }
+
+    this.setLastActive(this.timestamp); // update lastActive. User is active.
+  }
+
+  /**
+   * Persist session if page is focused.
+   */
+  private persistIfFocused(): void {
+    if (!document.hidden) {
+      this.persistSession();
+    }
+  }
+
+  /**
+   * Send POST to unique-events API endpoint
+   */
+  public async sendUniqueEvent(): Promise<void> {
+    const payload = {
+      application: this.applicationName,
+      appversion: this.applicationVersion,
+      metadata: {
+        dropped_events: 0,
+      },
+      unique_events: [
+        {
+          timestamp: currentTimestamp(),
+          attributes: this.eventAttributes,
+          unique: ['guid'],
+        },
+      ],
+    };
+
+    await post(this.summedEndpoint, payload);
+  }
+
+  /**
+   * Send POST to summed-events API endpoint
+   */
+  public async sendSummedEvent(metricGroup: string): Promise<void> {
+    const payload = {
+      application: this.applicationName,
+      appversion: this.applicationVersion,
+      metadata: {
+        dropped_events: 0,
+      },
+      summed_events: [
+        {
+          timestamp: currentTimestamp(),
+          metric_group: metricGroup,
+          attributes: this.eventAttributes,
+        },
+      ],
+    };
+
+    await post(this.summedEndpoint, payload);
+  }
+
+  private getEventAttributes(): {[index: string]: any} {
+    const reportAttributes = new BacktraceReport().toJson();
```

Comment:
I think it's not a good thing to do. Also annotations are complex object without object structure, so they don't belong to unique/summed events structure. Let's sync up on this to determine what we should do :)